graphistry/compute/ai/symbolic.py

```diff
@@ -0,0 +1,881 @@
+# methods to list all fields for indexes. 
+search index=yourindex| fieldsummary | table field 
+or 
+search index=yourindex | stats values(*) AS * | transpose | table column | rename column AS Fieldnames
+or 
+search index=yourindex | stats dc() as * | transpose 
+or 
+search index=yourindex | table *
+
+# List the size of lookup files with an SPL search.
+| rest splunk_server=local /services/data/lookup-table-files/
+| rename eai:acl.app as app 
+| table app title 
+| search NOT title IN (*.kmz) 
+| map maxsearches=990 search="| inputlookup $title$ 
+| eval size=0
+| foreach * [ eval size=size+coalesce(len('<<FIELD>>'),0), app=\"$app$\", title=$title$ | fields app title size]" 
+| stats sum(size) by app title
+| sort - sum(size)
+
+# Search for disabled AD accounts that have been re-enabled: This query searches for any AD accounts that have been disabled in the past 90 days and then re-enabled in the past 24 hours. 
+| search index=YOURINDEX EventCode IN (4725, 4722) earliest=-90d | eval account=mvindex(Account_Name,1) 
+
+# Search Common EventCodes (EventID’s) for Suspicious Behavior: This query searches for suspicious behavior in a Windows environment by looking at common EventCodes (EventID’s).
+| source index=YOURINDEX (EventCode IN (4688 OR 4624 OR 4625) AND SubjectLogonId!=0x3e7) earliest=-1d | eval account=mvindex(Account_Name,1)
+[Last Example]
+--------------
+"""
+
+
+class SplunkPreProcessor(PreProcessor):
+    def __call__(self, wrp_self, wrp_params, *args, **kwds):
+        super().override_reserved_signature_keys(wrp_params, *args, **kwds)
+        return "| {} SPL:".format(str(wrp_self))
+
+
+NEXT = " =>"
+
+
+class SplunkPrompts(ai.Prompt):
+    # https://docs.splunk.com/Documentation/SCS/current/SearchReference/SearchCommandExamples
+    def __init__(self) -> ai.Prompt:
+        super().__init__(
+            [
+                "search the main index where sourcetype is access_combined limit 10"
+                + NEXT
+                + "search index=main sourcetype=access_combined | head 10",
+                "Modify the following Splunk query that is useful for identifying malware in FireEye logs to instead use Palo Alto Networks logs:"
+                + NEXT
+                + "index=palo_alto sourcetype=pan:logs (action=allow OR action=deny) | eval src_ip=src_ip | eval dest_ip=dest_ip | where (src_ip=* OR dest_ip=*) | stats count by src_ip, dest_ip, action | sort - count",
+                "Lets try to find out how many errors have occurred on the Buttercup Games website"
+                + NEXT
+                + "buttercupgames (error OR fail* OR severe)",
+                "write a splunk query for the index `redteam_50k` that uses the src and dst information to output a table for events where RED=1. you can use closest matching fields from [src_computer, other, dst_computer, time]"
+                + NEXT
+                + '| search index="redteam_50k" RED=1 | Table src_computer, dst_computer',
+            ]
```

Comment:
Any guidance on tweaking these?
- We can bake in historic queries for a system
- Would it help to have 100 security related, or that's too many?
- ...